src/OrleansAzureUtils/Providers/Storage/AzureTableStorage.cs

```diff
@@ -246,49 +250,112 @@ internal void ConvertToStorageFormat(object grainState, GrainStateEntity entity)
 
                 if (Log.IsVerbose3) Log.Verbose3("Writing binary data size = {0} for grain id = Partition={1} / Row={2}",
                     data.Length, entity.PartitionKey, entity.RowKey);
-                
-                dataSize = data.Length;
-                entity.Data = data;
+
+                CheckMaxDataSize(data.Length, MAX_DATA_CHUNK_SIZE * MAX_DATA_CHUNKS_COUNT);
+
+                foreach (var partition in SplitBinaryData(data).Zip(GetBinaryDataPropertyNames(), (chunk, name) => new { chunk, name }))
+                {
+                    entity.Properties.Add(partition.name, new EntityProperty(partition.chunk));
```

Comment:
Yes, since `MAX_DATA_CHUNK_SIZE` is 64KB and `MAX_DATA_CHUNKS_COUNT` is 15, it makes `MAX_DATA_CHUNK_SIZE * MAX_DATA_CHUNKS_COUNT` 960 KB which should leave a room for PK, timestamp etc.